docs/codebase-best-practices.md

```diff
@@ -0,0 +1,61 @@
+# Codebase Best Practices
+
+## General JavaScript
+
+In most cases, our [linter](how-to-setup-freecodecamp-locally?id=follow-these-steps-to-get-your-development-environment-ready) will warn of any formatting which goes against this codebase's preferred practice.
+
+It is encouraged to use functional components over class-based components.
+
+## Specific TypeScript
+
+### Migrating a JavaScript File to TypeScript
+
+#### Retaining Git File History
+
+Sometimes changing the file from `<filename>.js` -> `<filename>.tsx?` causes the original file to be deleted, and a new one created, and other times the filename just changes - in terms of Git. Ideally, we want the file history to be preserved.
```

Comment:
```suggestion
Sometimes changing the file from `<filename>.js` -> `<filename>.tsx` causes the original file to be deleted, and a new one created, and other times the filename just changes - in terms of Git. Ideally, we want the file history to be preserved.
```
Not sure why there's a question mark here.

Reply:
The `x` is optional.

Reply:
How about:

```suggestion
Sometimes changing the file from `<filename>.js` to `<filename>.ts` (or `.tsx`) causes the original file to be deleted, and a new one created, and other times the filename just changes - in terms of Git. Ideally, we want the file history to be preserved.
```